### ISSUE TITLE
infinite recursion in MailHandler.emit_batch

### ISSUE THREAD

**Issue by yalon** [getlogbook/logbook#134]
Hi there,

In ```MailHandler``` there's ```emit_batch```, I think the following is a bug:
```python
    def emit_batch(self, records, reason):
        if reason not in ('escalation', 'group'):
            return MailHandler.emit_batch(self, records, reason)
        ...
```
If ```reason``` is not _escalation_ or _group_ we'll get an infinite recursive loop. I can create a pull request to fix this but we need to decide the correct behavior - I think in the case a ```RuntimeError``` should be raised since we don't have anything intelligent to do if passed an unrecognized reason.

**Comment by vmalloc** [comment]
Agreed.

**Comment by brainstorm** [comment]
+1, good catch @yalon!

**Comment by yalon** [comment]
Hi there, just posted a pull request for this (don't know why it doesn't appear in the issue).

**Comment by RazerM** [comment]
This can be closed, was fixed by c426f277445aa5277170255d5d992fbc33c6d1f3.

**Comment by brainstorm** [comment]
@yalon, @RazerM, indeed, it seems that it got merged in :+1:
